electron/new_config.ts

```diff
@@ -0,0 +1,294 @@
+import { exec } from 'child_process'
+import {
+  existsSync,
+  mkdir,
+  readFileSync,
+  readdirSync,
+  writeFileSync
+} from 'graceful-fs'
+import { userInfo as user } from 'os'
+
+import { AppSettings, ConfigVersion, WineProps } from './types'
+import {
+  currentConfigVersion,
+  heroicConfigPath,
+  heroicGamesConfigPath,
+  heroicInstallPath,
+  heroicToolsPath,
+  home,
+  userInfo
+} from './constants'
+import {
+  execAsync
+} from './utils'
+
+/**
+ * This class does config handling.
+ * This can't be constructed directly. Use the static method get().
+ * It automatically selects the appropriate config loader based on the config version.
+ *
+ * It also implements all the config features that won't change across versions.
+ */
+abstract class GlobalConfig {
+  protected static globalInstance : GlobalConfig
+
+  public abstract version : ConfigVersion
+
+  public config : AppSettings
+
+  /**
+   * Get the global configuartion handler.
+   *
+   * @param version ConfigVersion to use for parsing config file. Default: 'auto'
+   * @returns GlobalConfig instance or undefined.
+   */
+  public static get(version : ConfigVersion = 'auto') : GlobalConfig {
+    let configVersion : ConfigVersion = undefined
```

Comment:
Any specific reason o use the space before the `:`?

Reply:
Why start as undefined? can't it just start as 'auto'?

Reply:
No particular reason. I'll fix the colon.

Also I just start as `undefined` because that's convention for uninitialised variables. `auto` has a different meaning altogether.

Reply:
What convention is that? I never heard of it. All variables that do not start with a value, are undefined by default. 

I think you can just declare the variable or start with the default value that I thought was 'auto' or 'v0'.

Reply:
Will fix.